arctic/store/_ndarray_store.py

```diff
@@ -321,9 +321,27 @@ def _concat_and_rewrite(self, collection, version, symbol, item, previous_versio
                 'segment': {'$lt': version['up_to']}}
 
         read_index_range = [0, None]
-        unchanged_segment_ids = list(collection.find(spec, projection={'_id':1, 'segment':1},
-                                                     sort=[('segment', pymongo.ASCENDING)],))\
-                                                     [:-1 * (previous_version['append_count'] + 1)]
+        # The unchanged segments are those not yet compressed
+        unchanged_segment_ids = []
+        for segment in collection.find(spec, projection={'_id':1,
+                                                         'segment':1,
+                                                         'compressed': 1
+                                                         },
+                                       sort=[('segment', pymongo.ASCENDING)]):
+            # We want to stop iterating when we find the first uncompressed chunks
+            if not segment['compressed']:
+                # We include the last chunk in the recompression
+                unchanged_segment_ids.pop()
+                break
```

Comment:
I've added unit tests for this, and there's a test that asserts segment counts:
https://github.com/manahl/arctic/blob/master/tests/integration/store/test_ndarray_store.py#L200